### ISSUE TITLE
Train SDXL Lora with overtraining artifacts

### ISSUE THREAD

**Issue by KorDum** [bmaltais/kohya_ss#2152]
Hello! I have an strange issue with SDXL Lora's train.

I've tried both the newest version of Kohya_ss 23.0.15 (clean install without pip cache) and the oldest version 22.6.2 (clean install, too). The result is always the same.
I have also tried training on different video cards: RTX 3070 with 8 Gb and RTX 4090 with 24 Gb. There's no difference.
I always get overtraining artifacts on the first epoch (and next), no matter what learning rate I set.

The data set consists of images I generated that are 1024x1024 pixels in size.
I don't know what to do anymore. I've tried so many things. Can you help me please? I do not observe such problems with SD1.5.

![ComfyUI_temp_rqunk_00001_](https://github.com/bmaltais/kohya_ss/assets/1331559/46a09345-de01-4c7a-8c23-6e81b023c935)

```json
{
  "LoRA_type": "Standard",
  "LyCORIS_preset": "full",
  "adaptive_noise_scale": 0,
  "additional_parameters": "",
  "block_alphas": "",
  "block_dims": "",
  "block_lr_zero_threshold": "",
  "bucket_no_upscale": true,
  "bucket_reso_steps": 64,
  "cache_latents": true,
  "cache_latents_to_disk": true,
  "caption_dropout_every_n_epochs": 0.0,
  "caption_dropout_rate": 0,
  "caption_extension": ".txt",
  "clip_skip": 1,
  "color_aug": false,
  "constrain": 0.0,
  "conv_alpha": 1,
  "conv_block_alphas": "",
  "conv_block_dims": "",
  "conv_dim": 1,
  "debiased_estimation_loss": false,
  "decompose_both": false,
  "dim_from_weights": false,
  "down_lr_weight": "",
  "enable_bucket": true,
  "epoch": 1,
  "factor": -1,
  "flip_aug": false,
  "fp8_base": false,
  "full_bf16": true,
  "full_fp16": false,
  "gpu_ids": "",
  "gradient_accumulation_steps": 1,
  "gradient_checkpointing": true,
  "keep_tokens": "0",
  "learning_rate": 0.0012,
  "logging_dir": "F:/train/data/log",
  "lora_network_weights": "",
  "lr_scheduler": "constant",
  "lr_scheduler_args": "",
  "lr_scheduler_num_cycles": "",
  "lr_scheduler_power": "",
  "lr_warmup": 0,
  "max_bucket_reso": 2048,
  "max_data_loader_n_workers": "0",
  "max_grad_norm": 1,
  "max_resolution": "512,512",
  "max_timestep": 1000,
  "max_token_length": "75",
  "max_train_epochs": "",
  "max_train_steps": "",
  "mem_eff_attn": false,
  "mid_lr_weight": "",
  "min_bucket_reso": 256,
  "min_snr_gamma": 0,
  "min_timestep": 0,
  "mixed_precision": "bf16",
  "model_list": "custom",
  "module_dropout": 0,
  "multi_gpu": false,
  "multires_noise_discount": 0,
  "multires_noise_iterations": 0,
  "network_alpha": 1,
  "network_dim": 4,
  "network_dropout": 0,
  "noise_offset": 0,
  "noise_offset_type": "Original",
  "num_cpu_threads_per_process": 2,
  "num_machines": 1,
  "num_processes": 1,
  "optimizer": "Adafactor",
  "optimizer_args": "scale_parameter=False relative_step=False warmup_init=False",
  "output_dir": "D:/train/data/model",
  "output_name": "yajuu-xl-v2",
  "persistent_data_loader_workers": false,
  "pretrained_model_name_or_path": "F:\\StableDiffusion\\ComfyUI\\ComfyUI\\models\\checkpoints\\earthAnimixXLSemiflat_v15.safetensors",
  "prior_loss_weight": 1.0,
  "random_crop": false,
  "rank_dropout": 0,
  "rank_dropout_scale": false,
  "reg_data_dir": "",
  "rescaled": false,
  "resume": "",
  "sample_every_n_epochs": 0,
  "sample_every_n_steps": 0,
  "sample_prompts": "",
  "sample_sampler": "euler_a",
  "save_every_n_epochs": 1,
  "save_every_n_steps": 0,
  "save_last_n_steps": 0,
  "save_last_n_steps_state": 0,
  "save_model_as": "safetensors",
  "save_precision": "bf16",
  "save_state": false,
  "scale_v_pred_loss_like_noise_pred": false,
  "scale_weight_norms": 0,
  "sdxl": true,
  "sdxl_cache_text_encoder_outputs": false,
  "sdxl_no_half_vae": true,
  "seed": "",
  "shuffle_caption": false,
  "stop_text_encoder_training_pct": 0,
  "text_encoder_lr": 0.0012,
  "train_batch_size": 1,
  "train_data_dir": "F:/train/data/img",
  "train_norm": false,
  "train_on_input": true,
  "training_comment": "",
  "unet_lr": 0.0012,
  "unit": 1,
  "up_lr_weight": "",
  "use_cp": false,
  "use_scalar": false,
  "use_tucker": false,
  "use_wandb": false,
  "v2": false,
  "v_parameterization": false,
  "v_pred_like_loss": 0,
  "vae": "",
  "vae_batch_size": 0,
  "wandb_api_key": "",
  "weighted_captions": false,
  "xformers": "xformers"
}
```

**Comment by Mosfett1975** [comment]
if you get such a result in the first epoch, then of course you should not continue. Here's the json I'm training on right now
[adafactor.json](https://github.com/bmaltais/kohya_ss/files/14736803/adafactor.json)

**Comment by KorDum** [comment]
@Mosfett1975 thanks for ansfer!

I thought the problem was the bitsandbytes package. However, 0.41.1 and 0.43.0+ don't affect anything. Probably some dependency has been updated, but I don't know how to determine it.

**Comment by KorDum** [comment]
@Mosfett1975 Thanks for the config file, it works great!
I see a lot of discrepancies with mine - it remains to be seen which configuration is causing the problem. I'll report back here.

**Comment by Mosfett1975** [comment]
I've noticed now too - LORA is clearly not trained yet, but the background is starting to deteriorate. I've set up samples with a person in the city, the face doesn't look like it yet, but the paving stones are starting to disappear, the houses and the whole environment are blurred. I left in my set of photos only those with a simple white background, but it didn't help much

**Comment by Mosfett1975** [comment]
I figured out the background, the whole point is what pictures are at the entrance - the background was filled in black and in the captions I prescribed that the background is black :)

**Comment by KorDum** [comment]
@Mosfett1975 On a fully updated koha, including all pip dependencies? Is Triton enabled?

**Comment by bmaltais** [comment]
> @Mosfett1975 On a fully updated koha, including all pip dependencies? Is Triton enabled?

Not on Windows

**Comment by KorDum** [comment]
I get exactly the same results with retraining on Linux as I do on Windows. Only now 2.5 times faster :)
Still don't understand the reasoning behind it. Maybe the reason is that I'm not inputting real photos, but generated ones? I'm going to try to get someone's dataset with actual photos.

**Comment by KorDum** [comment]
Tried it on a third-party dataset, got exactly the same problems. I'm at a loss to guess what's going on. I have tried on two different computers with two different video cards and three different operating systems.

**Comment by v0xie** [comment]
Your config has max resolution specified as `"max_resolution": "512,512"`, but you also have `"sdxl": true`. 

You'd probably want to have the resolution set as 1024, 1024.

**Comment by KorDum** [comment]
@v0xie 
Unfortunately, my RTX 3070 doesn't allow me to specify this because 8GB VRAM. However, I also tried it on an RTX 4090 and 1024x1024. The outcome is the same.

**Comment by KorDum** [comment]
I tried latest version of Kohya_ss along with all dependency updates. The result is the same :(
I don't understand why this is happening.

**Comment by bmaltais** [comment]
Hard to tell. But if this set of parameters worked with a previous release then keep training with it. Create a new config that train well under the latest release. If probably is related to some of the sd-scripts code update that now require modifications of training parameters.

**Comment by HenryPotter0546** [comment]
### **Me too. I try to train a pixel style SDXL Lora using 150 pixel style images. The result shows good pixel style texture , but the outcome image shows color artifacts**

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/1500bcba-fd8e-4ccd-a018-6cac4f71a8ea" />

<img width="1024" height="1024" alt="Image" src="https://github.com/user-attachments/assets/5bbacbd1-3201-4b6e-b8fc-77f3515c520c" />
